src/conversions/uuid.rs

```diff
@@ -0,0 +1,301 @@
+use uuid::Uuid;
+
+use crate::conversion::IntoPyObject;
+use crate::exceptions::{PyTypeError, PyValueError};
+use crate::instance::Bound;
+use crate::sync::GILOnceCell;
+use crate::types::any::PyAnyMethods;
+use crate::types::bytearray::PyByteArrayMethods;
+use crate::types::{
+    IntoPyDict, PyByteArray, PyBytes, PyBytesMethods, PyInt, PyStringMethods, PyType,
+};
+use crate::{FromPyObject, Py, PyAny, PyErr, PyObject, PyResult, Python};
+#[allow(deprecated)]
+use crate::{IntoPy, ToPyObject};
+
+static UUID_CLS: GILOnceCell<Py<PyType>> = GILOnceCell::new();
+
+fn get_uuid_cls(py: Python<'_>) -> PyResult<&Bound<'_, PyType>> {
+    UUID_CLS.import(py, "uuid", "UUID")
+}
+
+impl FromPyObject<'_> for Uuid {
+    fn extract_bound(obj: &Bound<'_, PyAny>) -> PyResult<Self> {
+        let py = obj.py();
+
+        if let Ok(uuid_cls) = get_uuid_cls(py) {
+            if obj.is_exact_instance(&uuid_cls) {
+                let uuid_int: u128 = obj.getattr("int")?.extract()?;
+                return Ok(Uuid::from_u128(uuid_int.to_le()));
+            }
+        }
+
+        if obj.is_instance_of::<PyBytes>() || obj.is_instance_of::<PyByteArray>() {
+            let bytes = if let Ok(py_bytes) = obj.downcast::<PyBytes>() {
+                py_bytes.as_bytes()
+            } else if let Ok(py_bytearray) = obj.downcast::<PyByteArray>() {
+                &py_bytearray.to_vec()
+            } else {
+                return Err(PyTypeError::new_err(
+                    "Expected bytes or bytearray for UUID extraction.",
+                ));
+            };
+
+            return Uuid::from_slice(bytes)
+                .map_err(|_| PyValueError::new_err("The given bytes value is not a valid UUID."));
+        }
+
+        if obj.is_instance_of::<PyInt>() {
+            let uuid_int: u128 = obj.extract().map_err(|_| {
+                PyTypeError::new_err(
+                    "Expected integer for UUID extraction but got an incompatible type.",
+                )
+            })?;
+            return Ok(Uuid::from_u128(uuid_int));
+        }
+
+        let py_str = &obj.str()?;
+        let rs_str = &py_str.to_cow()?;
+        Uuid::parse_str(&rs_str)
+            .map_err(|e| PyValueError::new_err(format!("Invalid UUID string: {e}")))
+    }
+}
+
+#[allow(deprecated)]
+impl ToPyObject for Uuid {
+    #[inline]
+    fn to_object(&self, py: Python<'_>) -> PyObject {
+        self.into_pyobject(py).unwrap().into_any().unbind()
+    }
+}
+
+#[allow(deprecated)]
+impl IntoPy<PyObject> for Uuid {
+    #[inline]
+    fn into_py(self, py: Python<'_>) -> PyObject {
+        self.into_pyobject(py).unwrap().into_any().unbind()
+    }
+}
+
+impl<'py> IntoPyObject<'py> for Uuid {
+    type Target = PyAny;
+    type Output = Bound<'py, Self::Target>;
+    type Error = PyErr;
+
+    fn into_pyobject(self, py: Python<'py>) -> Result<Self::Output, Self::Error> {
+        let uuid_cls = get_uuid_cls(py)?;
+        let kwargs = [("int", self.as_u128())].into_py_dict(py)?;
```

Comment:
We are much more efficient calling as positional arguments, it might be worth benchmarking that here. (I guess the tuple would be something like `(None, None, None, None, int)`.)